src/Bicep.Core/Syntax/ModuleDeclarationSyntax.cs

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Bicep.Core.Navigation;
+using Bicep.Core.Parser;
+
+namespace Bicep.Core.Syntax
+{
+    public class ModuleDeclarationSyntax : SyntaxBase, IDeclarationSyntax
+    {
+        public ModuleDeclarationSyntax(Token keyword, IdentifierSyntax name, SyntaxBase path, SyntaxBase assignment, SyntaxBase body)
+        {
+            AssertKeyword(keyword, nameof(keyword), LanguageConstants.ModuleKeyword);
+            AssertSyntaxType(name, nameof(name), typeof(IdentifierSyntax));
+            AssertSyntaxType(path, nameof(path), typeof(StringSyntax), typeof(SkippedTriviaSyntax));
+            AssertTokenType(keyword, nameof(keyword), TokenType.Identifier);
+            AssertSyntaxType(assignment, nameof(assignment), typeof(Token), typeof(SkippedTriviaSyntax));
+            AssertTokenType(assignment as Token, nameof(assignment), TokenType.Assignment);
+
+            this.Keyword = keyword;
+            this.Name = name;
+            this.Path = path;
+            this.Assignment = assignment;
+            this.Body = body;
```

Comment:
I think we're missing the type assert here and in resources (`SkippedTriviaSyntax | ObjectSyntax` probably is sufficient)

Reply:
Added